File: Lecture_Code/Recursion/walk.cpp
#include <iostream>
using namespace std;

void walktogoal(int distance) {
    cout << "Entering walktogoal: with distance = " << distance << endl;
    if (distance == 0) {
        cout << "Reached goal" << endl;
        cout << "Exiting walktogoal: with distance = " << distance << endl;
        return;
    } else {
        cout << "Take one step " << endl;
        walktogoal(distance - 1);
        cout << "Exiting walktogoal: with distance = " << distance << endl;
    }
}
       


int main() {
    walktogoal(4);
}

File: Lecture_Code/Recursion/walk.py
def walktogoal(distance):
    if (distance == 0):
        print "Reached goal";
        return;
    else:
        print "Take one step";
        print "distance = ", distance;
        walktogoal(distance-1);
    

def buggywalk(distance):
    print "Take one step";
    print "distance = ", distance;
    buggywalk(distance-1);
